### ISSUE TITLE
Too much difference between Perceptual Loss, adversarial loss and L1 image loss

### ISSUE THREAD

**Issue by oubathAI** [xinntao/BasicSR#74]
Hi @xinntao 

I'm porting ESRGAN to Tensorflow using Keras.
When I've started ESRGAN training after training the first PSNR-oriented model, I've noticed that there is too much difference between the weigths of Perceptual Loss, Adversarial loss and L1 image loss using the loss scaler you gave. 

In fact, after the first epoch, I have Perceptual_loss=9.9663, Adversarial_loss=1.7964, IMG_loss=0.1070.

Is it intended  to have that much discrepancy at the begining of the training between weights, knowing that Adversarial and Img losses has been scaled down ? 

The good news is that Perceptual loss is decreasing at each epoch; should I just wait until it's go down to the same scale of the other losses?

I've also read SRGAN paper and remembered they scaled down Perceptual loss to have loss of same scale; and I wonder why for ESRGAN you didn't do it.

Thanks in advance for your answer

**Comment by xinntao** [comment]
1. when you have several losses, I think there is no need to let them be on the same scale. We may need to increase some loss weights to let them play a more important role during training. The loss weights for ESRGAN is from trials. 

2. Even in the final, the different losses could be on different scales. It does not matter. For adversarial loss, it may have other patterns, like increasing. 

3. There are many differences between SRGAN and ESRGAN, such as implementations and the used models (we use different pre-trained VGG models). So the loss weights may be different. 
I think there is no need to rescale one loss so that is comparable to the other losses. Because when do back-propagation. The gradients from different loss do matter rather than the loss values themselves.

**Comment by oubathAI** [comment]
Thanks for your clear answer !
